src/django_components/templatetags/component_tags.py

```diff
@@ -46,6 +60,71 @@
 register = django.template.Library()
 
 
+class TagSpec(NamedTuple):
```

Comment:
TagSpec is what was previously the input to the [`_parse_tag()`](https://github.com/EmilStenstrom/django-components/pull/716/files#diff-7345171ed38e1dd2e4b8199b65a2fd082323bdb331ff0ef0dc0dd3bd43f111aeR797) function. But it's now declared as standalone objects, so we can grab these objects when generating documentation.

I also found it confusing trying to remember what was the difference between `params` and `extra_params`. So I renamed the inputs and aligned them with how [Python represents function parameters](https://docs.python.org/3/library/inspect.html#inspect.Parameter.kind) the inputs, e.g. `positional_only_args`, `pos_or_keyword_args`, `keywordonly_args`.